lib/macho.rb

```diff
@@ -39,4 +40,22 @@ def self.open(filename)
 
     file
   end
+
+  # Signs the dylib using an ad-hoc identity.
+  # Necessary after making any changes to a dylib, since otherwise
+  # changing a signed file invalidates its signature.
+  # @param filename [String] the file being opened
+  # @return [void]
+  # @raise [ModificationError] if the operation fails
+  def self.codesign!(filename)
+    # codesign binary is not available on Linux
+    return if RUBY_PLATFORM !~ /darwin/
```

Comment:
Nit: I think it might make sense to raise an `UnsupportedError` or similar here instead of silently returning. I can take care of that this afternoon, if that sounds good to others.

Reply:
Thinking of having that be rescued in each of the `Tools` methods?

Reply:
Ah, I guess it would have to be, to avoid breaking those methods on non-macOS platforms 😞 

Hmm, that itself isn't ideal. I'm fine with leaving it as is for now 😅